src/UNL/Peoplefinder/Driver/OracleDB.php

```diff
@@ -280,6 +277,33 @@ public function fixLDAPEntries(array $entries)
             }
         }
 
+
+        //if we have BIODEMO_IDs we can then search for their affiliations
+        if(!empty($biodemo_id_values)){
+
+            $affiliation_query = "
+                SELECT UNL_AFFILIATIONS_02.BIODEMO_ID, UNL_AFFILIATIONS_02.NETID,
+                    LISTAGG(UNL_AFFILIATIONS_02.EPA_AFFILIATION, ';') WITHIN GROUP (ORDER BY UNL_AFFILIATIONS_02.EPA_AFFILIATION) as AFFILIATION
+                FROM UNL_AFFILIATIONS_02 
+                WHERE UNL_AFFILIATIONS_02.BIODEMO_ID IN (" . implode(', ', $biodemo_id_binding) . ")
+                GROUP BY UNL_AFFILIATIONS_02.BIODEMO_ID, UNL_AFFILIATIONS_02.NETID
+            ";
```

Comment:
Could we get much better performance by simplifying this down to a basic query like this?
SELECT UNL_AFFILIATIONS_02.EPA_AFFILIATION
	    FROM UNL_AFFILIATIONS_02
	    WHERE UNL_AFFILIATIONS_02.BIODEMO_ID IN {{ list of BIODEMO_IDs }}
	    
I had originally done the LISTAGG and grouping to cram this part into the bigger query, so that the affiliations would be a single value like student;staff - i think that was only needed to make it part of the larger query.

Reply:
then do all the sorting of the affiliations by person in PHP?

Reply:
I changed the query and aggregated the values in php